tsfresh/feature_extraction/extraction.py

```diff
@@ -190,32 +195,68 @@ def _extract_features_parallel_per_sample(kind_to_df_map, settings, column_id, c
                                                            column_value=column_value,
                                                            settings=settings)
     pool = Pool(settings.n_processes)
+    total_number_of_expected_results = 0
 
     # Submit map jobs per kind per sample
     results_fifo = Queue()
+
     for kind, df_kind in kind_to_df_map.items():
         df_grouped_by_id = df_kind.groupby(column_id)
+
+        total_number_of_expected_results += len(df_grouped_by_id)
+
+        chunksize = _calculate_best_chunksize(df_grouped_by_id, settings)
+
         results_fifo.put(
-            pool.map_async(
+            pool.imap(
                 partial_extract_features_for_one_time_series,
                 [(kind, df_group) for _, df_group in df_grouped_by_id],
-                chunksize=settings.chunksize
+                chunksize=chunksize
             )
         )
 
     pool.close()
 
     # Wait for the jobs to complete and concatenate the partial results
     dfs_per_kind = []
-    while not results_fifo.empty():
-        map_result = results_fifo.get()
-        dfs = map_result.get()
-        dfs_per_kind.append(pd.concat(dfs, axis=0).astype(np.float64))
 
-    result = pd.concat(dfs_per_kind, axis=1).astype(np.float64)
+    # Do this all with a progress bar
+    with tqdm(total=total_number_of_expected_results) as progress_bar:
+        # We need some sort of measure, when a new result is there. So we wrap the
+        # map_results into another iterable which updates the progress bar each time,
+        # a new result is there
+        def iterable_with_tqdm_update(queue, progress_bar):
+            for element in queue:
+                progress_bar.update(1)
+                yield element
 
-    pool.join()
-    return result
+        while not results_fifo.empty():
+            map_result = results_fifo.get()
+            dfs = iterable_with_tqdm_update(map_result, progress_bar)
+            dfs_per_kind.append(pd.concat(dfs, axis=0).astype(np.float64))
+
+        result = pd.concat(dfs_per_kind, axis=1).astype(np.float64)
+
+        pool.join()
+        return result
+
+
+def _calculate_best_chunksize(iterable_list, settings):
+    """
+    Helper function to calculate the best chunksize for a given number of elements to calculate,
+    or use the one in the settings object.
+    The chunksize calculation taken from map_async of multiprocessing module.
```

Comment:
Is this enough in terms of the Python license?

```
#
# Module providing the `Pool` class for managing a process pool
#
# multiprocessing/pool.py
#
# Copyright (c) 2006-2008, R Oudkerk
# All rights reserved.
#
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions
# are met:
#
# 1. Redistributions of source code must retain the above copyright
#    notice, this list of conditions and the following disclaimer.
# 2. Redistributions in binary form must reproduce the above copyright
#    notice, this list of conditions and the following disclaimer in the
#    documentation and/or other materials provided with the distribution.
# 3. Neither the name of author nor the names of any contributors may be
#    used to endorse or promote products derived from this software
#    without specific prior written permission.
#
# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS "AS IS" AND
# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
# SUCH DAMAGE.
#
```

I think we would have to reproduce the complete notice.